Assets/Code/Scripts/Level generation scripts/TileGenerationScript.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class TileGenerationScript : MonoBehaviour
+{
+    public int Size { get; set; }
+    
+    public List<GameObject> tiles;
+
+    public float Width { get; set; }
+
+    public int Blocks { get; set; }
+
+    public abstract GameObject Generate(int i, int j);
```

Comment:
minor detail: i think it would be good to give the variables more descriptive names than `i` and `j` just to avoid confusing them between themselves

Reply:
Sure I'll probably swap them to column and row